### ISSUE TITLE
Category not verified.  Need API About category. 

### ISSUE THREAD

**Issue by billychou** [jhuckaby/Cronicle#199]
## Summary

I create event  with  api `/api/app/create_event/v1`  request, then if the category is not valid,  the app crashed when job executed. 

Can you supply the event category api?  Thus i can verify the event。 Thanks. 

## Your Setup

version: 0.8.28
### Operating system and version?

CentOS 7.2 

### Node.js version?

`
$node --version
v8.9.1
`

### Are you using a multi-server setup, or just a single server?
a single server 
### Are you using the filesystem as back-end storage, or S3/Couchbase?
NAS filesystem 
### Can you reproduce the crash consistently?
YES
## Log Excerpts

```
TypeError: Cannot read property 'max_children' of null
    at /home/admin/tian-ji-cronicle/target/tian-ji-cronicle/lib/job.js:122:17
    at /home/admin/tian-ji-cronicle/target/tian-ji-cronicle/node_modules/_async@2.6.0@async/dist/async.js:3874:9
    at /home/admin/tian-ji-cronicle/target/tian-ji-cronicle/node_modules/_async@2.6.0@async/dist/async.js:473:16
    at replenish (/home/admin/tian-ji-cronicle/target/tian-ji-cronicle/node_modules/_async@2.6.0@async/dist/async.js:993:25)
    at iterateeCallback (/home/admin/tian-ji-cronicle/target/tian-ji-cronicle/node_modules/_async@2.6.0@async/dist/async.js:983:17)
    at /home/admin/tian-ji-cronicle/target/tian-ji-cronicle/node_modules/_async@2.6.0@async/dist/async.js:958:16
    at /home/admin/tian-ji-cronicle/target/tian-ji-cronicle/node_modules/_async@2.6.0@async/dist/async.js:3871:13
    at /home/admin/tian-ji-cronicle/target/tian-ji-cronicle/lib/job.js:102:6
    at /home/admin/tian-ji-cronicle/target/tian-ji-cronicle/node_modules/_pixl-server-storage@2.0.19@pixl-server-storage/list.js:630:7
    at /home/admin/tian-ji-cronicle/target/tian-ji-cronicle/node_modules/_async@2.6.0@async/dist/async.js:958:16
```

**Comment by jhuckaby** [comment]
Thanks for the bug report!

**Comment by regevbr** [comment]
Same here...

**Comment by regevbr** [comment]
For anyone trying to solve it - 
1. Stop the server
2. Set the `enabled` flag in `cronicle/global/state.json` to false
3. Start the server
4. Delete all events that has a bad category set on them
4. Enable the scheduling in the UI

@jhuckaby is there a quick way to avoid such an issue by enforcing the category exists upon creation of event (and also disable the ability to remove a category if there are events attached to it)? 
If it is straight forward I wouldn't mind setting a PR for it

**Comment by jhuckaby** [comment]
Working on this right now -- sorry for the delay.  Should have a new build by this weekend.

**Comment by jhuckaby** [comment]
Fixed in [v0.8.36](https://github.com/jhuckaby/Cronicle/releases/tag/v0.8.36).

> ...and also disable the ability to remove a category if there are events attached to it?

@regevbr Please note that this has always been the case.  You cannot delete any categories that have any events attached to them.  This is enforced both in the UI, and in the API.

**Comment by regevbr** [comment]
@jhuckaby thanks for the quick fix! I think you should also address the error that causes the entire server to crash. What do you think?

**Comment by jhuckaby** [comment]
@regevbr Sure, fixed in [v0.8.37](https://github.com/jhuckaby/Cronicle/releases/tag/v0.8.37).  Thanks.
